autogen/agentchat/contrib/vectordb/mongodb.py

```diff
@@ -0,0 +1,292 @@
+
+        Args:
+            queries: List[str] | A list of queries. Each query is a string.
+            collection_name: str | The name of the collection. Default is None.
+            n_results: int | The number of relevant documents to return. Default is 10.
+            distance_threshold: float | The threshold for the distance score, only distance smaller than it will be
+                returned. Don't filter with it if < 0. Default is -1.
+            kwargs: Dict | Additional keyword arguments.
+
+        Returns:
+            QueryResults | The query results. Each query result is a list of list of tuples containing the document and
+                the distance.
+        """
+        results = []
+        for query_text in queries:
+            query_vector = np.array(self.embedding_function([query_text])).tolist()[0]
+            # Find documents with similar vectors using the specified index
+            search_collection = self.get_collection(collection_name)
+            pipeline = [
+                {
+                    "$vectorSearch": {
+                        "index": index_name,
+                        "limit": n_results,
+                        "numCandidates": n_results,
+                        "queryVector": query_vector,
+                        "path": "embedding",
+                    }
+                },
+                {"$project": {"score": {"$meta": "vectorSearchScore"}}},
+            ]
+            if distance_threshold >= 0.00:
+                similarity_threshold = 1 - distance_threshold
+                pipeline.append({"$match": {"score": {"gte": similarity_threshold}}})
+
+            # do a lookup on the same collection
+            pipeline.append(
+                {
+                    "$lookup": {
+                        "from": collection_name,
+                        "localField": "_id",
+                        "foreignField": "_id",
+                        "as": "full_document_array",
+                    }
+                }
+            )
+            pipeline.append(
+                {
+                    "$addFields": {
+                        "full_document": {
+                            "$arrayElemAt": [
+                                {
+                                    "$map": {
+                                        "input": "$full_document_array",
+                                        "as": "doc",
+                                        "in": {"id": "$$doc.id", "content": "$$doc.content"},
+                                    }
+                                },
+                                0,
+                            ]
+                        }
+                    }
+                }
+            )
+            pipeline.append({"$project": {"full_document_array": 0, "embedding": 0}})
+            tmp_results = []
+            for doc in search_collection.aggregate(pipeline):
+                tmp_results.append((doc["full_document"], doc["score"]))
```

Comment:
`score` is the similarity, right? So, `1 - score` is what we need according to Protocol:
```
"""QueryResults is the response from the vector database for a query/queries.
A query is a list containing one string while queries is a list containing multiple strings.
The response is a list of query results, each query result is a list of tuples containing the document and the distance.
"""
QueryResults = List[List[Tuple[Document, float]]]
```

Reply:
Fixed!